internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/controller/TestIncrementalPush.java

```diff
@@ -1,110 +1,139 @@
     cluster = ServiceFactory.getVeniceCluster(
-        numberOfController,
-        numberOfServer,
-        numberOfRouter,
-        replicaFactor,
-        partitionSize,
-        false,
-        false);
+        new VeniceClusterCreateOptions.Builder().numberOfServers(3)
+            .replicationFactor(3)
+            .partitionSize(PARTITION_SIZE)
+            .build());
   }
 
-  @AfterMethod
-  public void cleanUp() {
+  @AfterClass(alwaysRun = true)
+  public void tearDownClass() {
     cluster.close();
   }
 
-  @Test(timeOut = 120 * Time.MS_PER_SECOND)
-  public void testGetOfflineStatusIncrementalPush() {
-    String storeName = Utils.getUniqueString("testIncPushStore");
-    int partitionCount = 2;
-    long storageQuota = (long) partitionCount * partitionSize;
+  @BeforeMethod(alwaysRun = true)
+  public void setUp() {
+    storeName = Utils.getUniqueString("testIncPushStore");
     cluster.getNewStore(storeName);
+    long storageQuota = 2L * PARTITION_SIZE;
     UpdateStoreQueryParams params = new UpdateStoreQueryParams();
     params.setIncrementalPushEnabled(true)
         .setHybridRewindSeconds(1)
         .setHybridOffsetLagThreshold(1)
-        .setStorageQuotaInByte(storageQuota);
+        .setStorageQuotaInByte(storageQuota)
+        .setBackupStrategy(BackupStrategy.KEEP_MIN_VERSIONS)
+        .setNumVersionsToPreserve(3);
     cluster.updateStore(storeName, params);
+  }
 
-    VersionCreationResponse response = cluster.getNewVersion(storeName);
-    Assert.assertFalse(response.isError());
-    String topicName = response.getKafkaTopic();
-    // push version 1 to completion
-    VeniceWriter<String, String, byte[]> veniceWriter = cluster.getVeniceWriter(topicName);
-    veniceWriter.broadcastStartOfPush(new HashMap<>());
-    veniceWriter.broadcastEndOfPush(new HashMap<>());
+  @AfterMethod(alwaysRun = true)
+  public void cleanUp() {
+    cluster.useControllerClient(controllerClient -> controllerClient.deleteStore(storeName));
+  }
 
-    // start inc push
-    String incPushVerison = "incPush1";
-    veniceWriter.broadcastStartOfIncrementalPush(incPushVerison, new HashMap<>());
-    veniceWriter.broadcastEndOfIncrementalPush(incPushVerison, new HashMap<>());
+  @Test(timeOut = 2 * Time.MS_PER_MINUTE)
+  public void testGetOfflineStatusIncrementalPush() {
+    // store version 1
+    VersionCreationResponse v1Response = cluster.getNewVersion(storeName);
+    Assert.assertFalse(v1Response.isError());
+    String versionTopic1 = v1Response.getKafkaTopic();
+    // push version 1 to completion
+    VeniceWriter<String, String, byte[]> veniceWriterVt1 = cluster.getVeniceWriter(versionTopic1);
+    veniceWriterVt1.broadcastStartOfPush(new HashMap<>());
+    veniceWriterVt1.broadcastEndOfPush(new HashMap<>());
+    // write incremental push to store version 1
+    String incPushV1 = "incPush1";
+    veniceWriterVt1.broadcastStartOfIncrementalPush(incPushV1, new HashMap<>());
```

Comment:
Why do we still write SOIP into VT?

Reply:
To populate VT with incremental push  version. This allows us to skip pushing data from clients